tools/lambda-promtail/lambda-promtail/kinesis_test.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"testing"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/stretchr/testify/require"
+)
+
+type MockBatch struct {
+	streams map[string]*logproto.Stream
+	size    int
+}
+
+func (b *MockBatch) add(ctx context.Context, e entry) error {
+	return nil
+}
+
+func (b *MockBatch) flushBatch(ctx context.Context) error {
+	return nil
+}
+func (b *MockBatch) encode() ([]byte, int, error) {
+	return nil, 0, nil
+}
+func (b *MockBatch) createPushRequest() (*logproto.PushRequest, int) {
+	return nil, 0
+}
+
+func ReadJSONFromFile(t *testing.T, inputFile string) []byte {
+	inputJSON, err := ioutil.ReadFile(inputFile)
+	if err != nil {
+		t.Errorf("could not open test file. details: %v", err)
+	}
+
+	return inputJSON
+}
+
+func TestLambdaPromtail_KinesisParseEvents(t *testing.T) {
+	inputJson, err := ioutil.ReadFile("../testdata/kinesis-event.json")
+
+	if err != nil {
+		t.Errorf("could not open test file. details: %v", err)
+	}
+
+	var testEvent events.KinesisEvent
+	if err := json.Unmarshal(inputJson, &testEvent); err != nil {
+		t.Errorf("could not unmarshal event. details: %v", err)
+	}
+
+	ctx := context.TODO()
+	b := &MockBatch{
+		streams: map[string]*logproto.Stream{},
+	}
+
+	err = parseKinesisEvent(ctx, b, &testEvent)
+
+	require.Nil(t, err)
```

Comment:
I think we should check here that the labels you're extracting actually get extracted as expected?

Reply:
Added a check for the labels.